components/ntp_background_images/browser/ntp_background_images_source_unittest.cc

```diff
@@ -43,35 +43,46 @@ class NTPBackgroundImagesSourceTest : public testing::Test {
 };
 
 TEST_F(NTPBackgroundImagesSourceTest, SponsoredImagesTest) {
-  const std::string test_json_string_referral = R"(
-      {
-        "schemaVersion": 1,
-        "logo": {
-          "imageUrl": "logo.png",
-          "alt": "Technikke: For music lovers",
-          "companyName": "Technikke",
-          "destinationUrl": "https://www.brave.com/?from-super-referreer-demo"
-        },
-        "wallpapers": [
-          {
-            "imageUrl": "background-1.jpg",
-            "focalPoint": { "x": 3988, "y": 2049}
-          },
-          {
-            "imageUrl": "background-2.jpg",
-            "focalPoint": { "x": 5233, "y": 3464}
-          },
-          {
-            "imageUrl": "background-3.jpg"
-          }
-        ]
-      })";
+  const std::string test_json_string_referral = R"JSON(
+    {
+      "schemaVersion": 2,
+      "campaigns": [
+        {
+          "version": 1,
+          "campaignId": "65933e82-6b21-440b-9956-c0f675ca7435",
+          "creativeSets": [
+            {
+              "creativeSetId": "6690ad47-d0af-4dbb-a2dd-c7a678b2b83b",
+              "creatives": [
+                {
+                  "creativeInstanceId": "30244a36-561a-48f0-8d7a-780e9035c57a",
+                  "companyName": "Image NTT Creative",
+                  "alt": "Some content",
+                  "targetUrl": "https://basicattentiontoken.org",
+                  "wallpaper": {
+                    "type": "image",
+                    "relativeUrl": "6690ad47-d0af-4dbb-a2dd-c7a678b2b83b/background.jpg",
+                    "focalPoint": {
+                      "x": 25,
+                      "y": 50
+                    },
+                    "button": {
+                      "image": {
+                        "relativeUrl": "6690ad47-d0af-4dbb-a2dd-c7a678b2b83b/button.png"
+                      }
+                    }
+                  }
+                }
+              ]
+            }
+          ]
+        }
+      ]
+    })JSON";
   service_->OnGetSponsoredComponentJsonData(false, test_json_string_referral);
   EXPECT_FALSE(source_->AllowCaching());
-  EXPECT_TRUE(source_->IsValidPath("sponsored-images/logo.png"));
-  EXPECT_TRUE(source_->IsValidPath("sponsored-images/background-1.jpg"));
-  EXPECT_TRUE(source_->IsValidPath("sponsored-images/background-2.jpg"));
-  EXPECT_TRUE(source_->IsValidPath("sponsored-images/background-3.jpg"));
+  EXPECT_TRUE(source_->IsValidPath("sponsored-images/button.png"));
+  EXPECT_TRUE(source_->IsValidPath("sponsored-images/background.jpg"));
```

Comment:
you're not checking any invalid paths?

Reply:
also per comment above we should check paths with `..` that could potentially read outside of the base directory

Reply:
This was only updating the filenames to match the new JSON structure. This PR is for RichNTT and not NTP backgrounds in general.

Reply:
Added other tests which will cover this until NTT background/sponsored images are refactored.